bids/analysis/transformations/compute.py

```diff
@@ -138,9 +138,19 @@ class Scale(Transformation):
         None, no replacement is performed. If 'before', missing values are
         replaced with 0's before scaling. If 'after', missing values are
         replaced with 0 after scaling.
+
+    Notes
+    -----
+    If a constant column is passed in, and replace_na is None, an exception
+    will be raised.
     """
 
     def _transform(self, data, demean=True, rescale=True, replace_na=None):
+        if data.nunique() == 1 and replace_na is None:
```

Comment:
Why only when `replace_na` is None?

Reply:
Because if you explicitly say you want NaNs replaced, I think that should still happen, otherwise you'd be getting back NaNs when you don't want any.

Reply:
I suppose if you pass `'before'` you'll still get back NaNs, so yeah, that needs to be updated.